src/main/java/mate/academy/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Order;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+
+    @Override
+    public Order add(Order order) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(order);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can`t add order: " + order, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return order;
+    }
+
+    @Override
+    public List<Order> getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("FROM Order o "
+                    + "LEFT JOIN fetch o.user u "
+                    + "LEFT JOIN fetch o.tickets t "
+                    + "LEFT JOIN fetch t.movieSession ms "
+                    + "LEFT JOIN fetch ms.cinemaHall ch "
+                    + "LEFT JOIN fetch ms.movie m "
+                    + "WHERE o.user =:user", Order.class)
```

Comment:
here the same. in this case an order can't exist without tickets, or user, that why we can use simple join. otherwise (for example shopping cart can exist without tickets) we need to use left join